Apps/Src/FaultState.c

```diff
@@ -93,8 +94,18 @@ void EnterFaultState(void) {
 
          */
     }
+    else if(FaultBitmap & FAULT_BPS){ // BPS has been tripped
+
+        /* TODO: set display to show BPS has been tripped */
+
+        nonrecoverableFaultHandler();
+        return;
+    }
     else if(FaultBitmap & FAULT_READBPS){ // Missed BPS Charge enable message
         readBPS_ContactorHandler();
+
+        // Reset FaultBitmap since this is a recoverable fault
+        FaultBitmap = FAULT_NONE; // Clears everything because only one error can be set at a time
```

Comment:
Fault bitmap can actually hold multiple errors, since it's a bitmap and each bit represents a different error, so you may only want to clear one bit (the FAULT_READBPS bit in this case)

Reply:
Sorry, that one's on me. My impression was that if we return from fault state, then it means we shouldn't have any faults still asserted in the bitmap. @babusid thoughts?

Reply:
If two faults trigger at the same time then we need to go back into fault state to handle the second fault right? Speaking of which, is there an edge case there where a higher priority fault doesn't get handled quickly enough because of something like a display error?